client/src/actions/teaActions.js

```diff
@@ -1,94 +1,42 @@
-import { RSAA } from "redux-api-middleware";
 import API from "../lib/api";
-
-let backendHost;
-const hostname = window && window.location && window.location.hostname;
-
-if (hostname === "localhost") {
-  backendHost = "http://localhost:5000";
-} else {
-  backendHost = "";
-}
-const API_SERVER = `${backendHost}`;
+import teaNormalizer from "./normalizers/teaNormalizer";
 
 // Add Tea
 export function addTea(tea) {
-  console.log("this one");
-  tea.id = tea.teaId;
-  return {
-    [RSAA]: {
-      endpoint: `${API_SERVER}/api/teas/new-tea`,
-      method: "POST",
-      types: [
-        "REQUEST",
-        {
-          type: "ADD_TEA",
-          payload: async (_action, _state, res) => {
-            res = await res.json();
-            return res;
-          }
-        },
-        {
-          type: "GET_ERRORS",
-          payload: async (_action, _state, res) => res.response.data
-        }
-      ],
-      body: JSON.stringify(tea),
-      headers: { "Content-Type": "application/json" }
-    }
+  tea.id = tea.teaID;
+  console.log(tea);
+  return dispatch => {
+    API.post(`/teas/new-tea`, tea).then(response => {
+      dispatch({
+        type: "EDIT_TEA",
+        payload: teaNormalizer(response)
```

Comment:
I would usually suggest performing normalization within your reducer. There's the off chance that some other reducer might want the entire object. If you don't think that will be the case, you can save some front-end work by doing the normalization on the back-end before sending your response.